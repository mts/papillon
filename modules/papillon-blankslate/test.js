import React from 'react';
import {
  blankslate,
  blankslateIcon,
  blankslateNarrow,
} from './flavor';

describe('<Blankslate />', () => {
  let wrapper;

  describe('Instance', () => {
    test('must be an instance of Blankslate', () => {
      wrapper = renderMount(blankslate);
      expect(wrapper.find('Blankslate')).toHaveLength(1);
    });
  });

  describe('Snaphot', () => {
    test('must match blankslate', () => {
      expect(global.renderToJSON(blankslate)).toMatchSnapshot();
    });

    test('must match blankslateIcon', () => {
      expect(global.renderToJSON(blankslateIcon)).toMatchSnapshot();
    });

    test('must match blankslateNarrow', () => {
      expect(global.renderToJSON(blankslateNarrow)).toMatchSnapshot();
    });
  });
});
