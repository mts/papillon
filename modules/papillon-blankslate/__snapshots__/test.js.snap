// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blankslate /> Snaphot must match blankslate 1`] = `
<div
  className="blankslate"
>
  <h3>
    This is a blank slate
  </h3>
  <p>
    Use it to provide information when no dynamic content exists.
  </p>
</div>
`;

exports[`<Blankslate /> Snaphot must match blankslateCapped 1`] = `
<div
  className="blankslate blankslate-capped"
>
  <h3>
    This is a blank slate
  </h3>
  <p>
    Use it to provide information when no dynamic content exists.
  </p>
</div>
`;

exports[`<Blankslate /> Snaphot must match blankslateCleanBackground 1`] = `
<div
  className="blankslate blankslate-clean-background"
>
  <h3>
    This is a blank slate
  </h3>
  <p>
    Use it to provide information when no dynamic content exists.
  </p>
</div>
`;

exports[`<Blankslate /> Snaphot must match blankslateIcon 1`] = `
<div
  className="blankslate"
>
  <span
    className="blankslate-icon"
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 14 16"
      width={14}
    >
      <path
        d="M10.86 7c-.45-1.72-2-3-3.86-3-1.86 0-3.41 1.28-3.86 3H0v2h3.14c.45 1.72 2 3 3.86 3 1.86 0 3.41-1.28 3.86-3H14V7h-3.14zM7 10.2c-1.22 0-2.2-.98-2.2-2.2 0-1.22.98-2.2 2.2-2.2 1.22 0 2.2.98 2.2 2.2 0 1.22-.98 2.2-2.2 2.2z"
        fillRule="evenodd"
      />
    </svg>
  </span>
  <span
    className="blankslate-icon"
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 14 16"
      width={14}
    >
      <path
        d="M7.73 1.73C7.26 1.26 6.62 1 5.96 1H3.5C2.13 1 1 2.13 1 3.5v2.47c0 .66.27 1.3.73 1.77l6.06 6.06c.39.39 1.02.39 1.41 0l4.59-4.59a.996.996 0 0 0 0-1.41L7.73 1.73zM2.38 7.09c-.31-.3-.47-.7-.47-1.13V3.5c0-.88.72-1.59 1.59-1.59h2.47c.42 0 .83.16 1.13.47l6.14 6.13-4.73 4.73-6.13-6.15zM3.01 3h2v2H3V3h.01z"
        fillRule="evenodd"
      />
    </svg>
  </span>
  <span
    className="blankslate-icon"
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 10 16"
      width={10}
    >
      <path
        d="M10 5c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v.3c-.02.52-.23.98-.63 1.38-.4.4-.86.61-1.38.63-.83.02-1.48.16-2 .45V4.72a1.993 1.993 0 0 0-1-3.72C.88 1 0 1.89 0 3a2 2 0 0 0 1 1.72v6.56c-.59.35-1 .99-1 1.72 0 1.11.89 2 2 2 1.11 0 2-.89 2-2 0-.53-.2-1-.53-1.36.09-.06.48-.41.59-.47.25-.11.56-.17.94-.17 1.05-.05 1.95-.45 2.75-1.25S8.95 7.77 9 6.73h-.02C9.59 6.37 10 5.73 10 5zM2 1.8c.66 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2C1.35 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2zm0 12.41c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm6-8c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
        fillRule="evenodd"
      />
    </svg>
  </span>
  <h3>
    This is a blank slate
  </h3>
  <p>
    Use it to provide information when no dynamic content exists.
  </p>
</div>
`;

exports[`<Blankslate /> Snaphot must match blankslateLarge 1`] = `
<div
  className="blankslate blankslate-large"
>
  <h3>
    This is a blank slate
  </h3>
  <p>
    Use it to provide information when no dynamic content exists.
  </p>
</div>
`;

exports[`<Blankslate /> Snaphot must match blankslateNarrow 1`] = `
<div
  className="blankslate blankslate-narrow"
>
  <h3>
    This is a blank slate
  </h3>
  <p>
    Use it to provide information when no dynamic content exists.
  </p>
</div>
`;
